packages/babel-traverse/src/path/family.js

```diff
@@ -67,6 +67,39 @@ export function getSibling(key) {
   });
 }
 
+export function getPrevSibling() {
+  return this.getSibling(this.key - 1);
+}
+
+export function getNextSibling() {
+  return this.getSibling(this.key + 1);
+}
+
+export function getAllNextSibling(callback, thisArg = this) {
+  let _key = this.key;
+  let sibling = this.getSibling(++_key);
+  let siblings = [];
+  const hasCallback = callback && typeof callback === "function";
+  while (sibling.node) {
+    siblings = siblings.concat(hasCallback ? callback.call(thisArg, sibling) : sibling);
+    sibling = this.getSibling(++_key);
+  }
+  return siblings;
+}
+
+export function getAllPrevSibling(callback, thisArg = this) {
+  let _key = this.key;
+  let sibling = this.getSibling(--_key);
+  let siblings = [];
+  const hasCallback = callback && typeof callback === "function";
+  while (sibling.node) {
+    siblings = siblings.concat(hasCallback ? callback.call(thisArg, sibling) : sibling);
```

Comment:
Why not just push to array? Overwriting and concating array every time is almost definitely slower.